src/transformers/pipelines/base.py

```diff
@@ -1035,10 +1036,23 @@ def forward(self, model_inputs, **forward_params):
     def get_iterator(
         self, inputs, num_workers: int, batch_size: int, preprocess_params, forward_params, postprocess_params
     ):
+        try:
+            n = len(inputs)
+        except TypeError:
```

Comment:
In the `Trainer` we use `isinstance(self.train_dataset, collections.abc.Sized)` to determine if the training dataset has a length or not.

Reply:
Thanks makes code better !